RadixWallet/Clients/URLFormatterClient/URLFormatterClient+Live.swift

```diff
@@ -23,10 +23,30 @@ extension URLFormatterClient: DependencyKey {
 	#endif
 
 	private static func makeURL(url: URL, imageServiceURL: URL, size: CGSize) -> URL {
-		let originItem = URLQueryItem(name: "imageOrigin", value: url.absoluteString)
-		let sizeItem = URLQueryItem(name: "imageSize", value: imageSizeString(size: size))
+		if url.isDataURL {
+			// Data URLs (e.g., `data:image/svg+xml,`) require special encoding to handle special characters.
+			// Here, we encode the absolute string using percent encoding, allowing only alphanumeric characters.
+			// This is needed because `url.appending(queryItems:)` does not encode special characters correctly for data URLs.
+			let imageOrigin = url.absoluteString.addingPercentEncoding(withAllowedCharacters: .alphanumerics) ?? ""
+			let imageSize = imageSizeString(size: size)
+
+			// Manually construct the query string to ensure that the data URL is correctly encoded
```

Comment:
We would probably benefit from having this written in Sargon. WDYT?

Reply:
Yeah, especially with `isVectorImage(type:)`. Android will also need to check if an image is an SVG.